### PR TITLE
Different key handling, fixes accents/dead-keys and alt-shortcuts problems (#12)

### PR DESCRIPTION
**Pull request by ikeblaster**
**TLDR:** new way of detecting when surrounding should be done. It has flaws, but works perfectly on US and CZ keyboard layouts. 
Switching layouts is probably broken (not part of this code, was broken before) - you have to restart N++ to fix which keys are used for surrounding text.

---

In initialization and every KBD layout change* it loops over all surrounding characters (``"'`(){}[]<>`` + `AdditionalChars` - `IgnoreChars`) and prepares an array of keyboard state combination on which surrounding should be triggered.

I.e. for `"` it will add one `COMBINATION` struct with virtual key code and shift/alt/ctrl states needed to produce `"` on current keyboard layout.

Then in KBD procedure it loops through these combinations. If any of them matches current keyboard state, surrounding is triggered.

\* This probably doesn't work. I think `messageProc` is not processed inside plugins so at the moment it won't detect any keyboard layout changes.

---

Sadly it's still flawed and I am a bit afraid how it will work for different keyboard layouts. There are keys, which have more than one way to type. And function `VkKeyScanEx` just chooses one of them.

For example it's not possible to use `´` key (next to backspace on my keyboard) in `AdditionalChars`, because `´` character is actually translated into `ctrl+alt+9` (key above O). If you press this combination, you will get the text surrounded, but not if you press that separate `´` key.

Workaround would be to do some heavy processing beforehand - analyzing whole keyboard and finding all the combinations to type some character. And then use these combinations instead of relying on `VkKeyScanEx`.

Good source explaining all this: http://archives.miloush.net/michkap/archive/2008/09/03/8921419.html

---

I tried US KB layout and all the default characters work. Also on CZ layout. So hopefully the default characters will be okay for everyone. And I think it's still better than the previous version, which broke dead keys completely (because of flaw in another WinAPI function... it's a real pleasure to work with keyboard processing, thanks Microsoft).

### CODE REVIEW

**Comment by ikeblaster** [comment]
Here are binaries if anyone wants to try it. I tested only 32bit version.

[SurroundSelection.patch-accents.zip](https://github.com/dail8859/SurroundSelection/files/4035277/SurroundSelection.patch-accents.zip)

**Comment by ikeblaster** [comment]
Apparently also fixes #10

**Comment by dail8859** [comment]
Greatly appreciate the pull request!

Have a pretty busy schedule for the time being but will get around to reviewing this as soon as I have a bit of free time.

**Comment by dail8859** [comment]
> I think it's still better than the previous version

Yes this is the key point. I am sure doing it *completely* correct is non-trivial so as long as it is an improvement then that is definitely a good thing even if it isn't perfect.

Did some testing and everything seems to be working fine for me!

Thanks again for the PR. Will squash and merge this momentarily.
